### PR TITLE
[docs] EAS Builds: add 'Setup' page & more

### PR DESCRIPTION
**Pull request by dsokal**
# Why

This is probably the last part of adding docs for EAS Builds. At least for now :)

Previous PRs: #9566, #9613, #9704, and #9721

# How

- I added a new page for EAS Builds - "Setup"
- I updated the "Introduction" page.
- I addressed Brent's suggestions with the last PR - using `/* @info */` for highlighting some parts of code.

# Test Plan

I ran the dev server and made sure the new page renders correctly.

### CODE REVIEW

**Review by wkozyra95 — APPROVED** [review]

File: docs/pages/build/eas-builds-in-5-minutes.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ Let's get started by building Android and iOS app binaries for a fresh bare proj
 
 ## Prerequisites
 
-- Install Expo CLI by running `npm install -g expo-cli` (or `yarn global add expo-cli`). _If you already have it, make sure you're using the latest version._ EAS Builds is in alpha and it's changing rapidly, so the only way to ensure that you will have a good experience is to use the latest expo-cli version. **This tutorial assumes you're using `expo-cli@3.23.2` or newer.**
+- Install Expo CLI by running `npm install -g expo-cli` (or `yarn global add expo-cli`). _If you already have it, make sure you're using the latest version._ EAS Builds is in alpha and it's changing rapidly, so the only way to ensure that you will have a good experience is to use the latest expo-cli version. **This tutorial assumes you're using `expo-cli@3.24.0` or newer.**
```
Comment (cruzach) [inline docs/pages/build/eas-builds-in-5-minutes.md:9]:
```suggestion
- Install Expo CLI by running `npm install -g expo-cli` (or `yarn global add expo-cli`). _If you already have it, make sure you're using the latest version._ EAS Builds is in alpha and it's changing rapidly, so the only way to ensure that you will have the best experience is to use the latest expo-cli version. **This tutorial assumes you're using `expo-cli@3.24.0` or newer.**
```

File: docs/pages/build/eas-builds-in-5-minutes.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ Let's get started by building Android and iOS app binaries for a fresh bare proj
...
 - Sign in with `expo login` or sign up with `expo register` if you don't have an Expo account yet. You can check if you're logged in by running `expo whoami`.
```
Comment (cruzach) [inline docs/pages/build/eas-builds-in-5-minutes.md:10]:
```suggestion
- Sign in with `expo login`, or sign up with `expo register` if you don't have an Expo account yet. You can check if you're logged in by running `expo whoami`.
```

File: docs/pages/build/eas-json.md [hunk 3]
```diff
@@ -67,7 +67,7 @@ The schema of a build profile for a generic Android project looks like this:
 ```
 
 - `"workflow": "generic"` indicates that your project is a generic one.
-- `credentialsSource` defines the credentials source for the project build. If you want to take advantage of your own `credentials.json` file, set it to `local` ([learn more on this here](../advanced-credentials/)). If you always want to use credentials stored on Expo servers, choose `remote`. If you're not sure what to do but you probably won't be needing to run builds from a CI, choose `auto` (this is the default option).
+- `credentialsSource` defines the credentials source for the project build. If you want to take advantage of your own `credentials.json` file, set it to `local` ([learn more on this here](../advanced-credentials-configuration/)). If you always want to use credentials stored on Expo servers, choose `remote`. If you're not sure what to do but you probably won't be needing to run builds from a CI, choose `auto` (this is the default option).
```
Comment (cruzach) [inline docs/pages/build/eas-json.md:70]:
```suggestion
- `credentialsSource` defines the source of credentials for this build profile. If you want to take advantage of your own `credentials.json` file, set this to `local` ([learn more on this here](../advanced-credentials-configuration/)). If you want to use the credentials Expo already has stored for you, choose `remote`. If you're not sure what to do but you probably won't be running builds from a CI, choose `auto` (this is the default option).
```

File: docs/pages/build/eas-json.md [hunk 4]
```diff
@@ -130,7 +130,7 @@ The schema of a build profile for a generic iOS project looks like this:
...
```
Comment (cruzach) [inline docs/pages/build/eas-json.md:133]:
```suggestion
- `credentialsSource` defines the source of credentials for this build profile. If you want to take advantage of your own `credentials.json` file, set this to `local` ([learn more on this here](../advanced-credentials-configuration/)). If you want to use the credentials Expo already has stored for you, choose `remote`. If you're not sure what to do but you probably won't be running builds from a CI, choose `auto` (this is the default option).
```

File: docs/pages/build/eas-json.md [hunk 5]
```diff
@@ -143,12 +143,12 @@ This is the minimal working example. Expo CLI will ask you for the app's credent
 }
 ```
 
-If you'd like to build your iOS project with your custom `Gymfile` ([learn more on this here](../ios/)) and you've defined a different output directory than `ios/build` use the following example:
+If you'd like to build your iOS project with your custom `Gymfile` ([learn more on this here](../ios-builds/#building-ios-projects-with-fastlane)) and you've defined a different output directory than `ios/build` use the following example:
```
Comment (cruzach) [inline docs/pages/build/eas-json.md:146]:
```suggestion
If you'd like to build your iOS project with a custom `Gymfile` ([learn more on this here](../ios-builds/#building-ios-projects-with-fastlane)) where you've defined a different output directory than `ios/build`, use the following example:
```

File: docs/pages/build/setup.md [hunk 6]
```diff
@@ -2,4 +2,72 @@
 title: Setup
 ---
 
-Here is the basic setup, eg: create eas.json. If using managed project, eject. Whatever else we want to put here.
\ No newline at end of file
+Follow these instructions to get ready for using EAS Builds.
+
+## 1. Install Expo CLI
+
+Install Expo CLI by running `npm install -g expo-cli` (or `yarn global add expo-cli`). If you already have it, make sure you're using the latest version. EAS Builds is in alpha and it's changing rapidly, so the only way to ensure that you will have a good experience is to use the latest expo-cli version.
```
Comment (cruzach) [inline docs/pages/build/setup.md:9]:
```suggestion
Install Expo CLI by running `npm install -g expo-cli` (or `yarn global add expo-cli`). If you already have it, make sure you're using the latest version. EAS Builds is in alpha and it's changing rapidly, so the only way to ensure that you will have the best experience is to use the latest expo-cli version.
```

File: docs/pages/build/setup.md [hunk 7]
```diff
@@ -2,4 +2,72 @@
...
+
+## 2. Sign In
+
+Sign in with `expo login` or sign up with `expo register` if you don't have an Expo account yet. You can check if you're logged in by running `expo whoami`.
```
Comment (cruzach) [inline docs/pages/build/setup.md:13]:
```suggestion
Sign in with `expo login`, or sign up with `expo register` if you don't have an Expo account yet. You can check if you're logged in by running `expo whoami`.
```

File: docs/pages/build/setup.md [hunk 8]
```diff
@@ -2,4 +2,72 @@
...
+
+**⚠️ Do not follow this step if you are using the bare workflow or a vanilla React Native project.**
+
+Building managed Expo projects with EAS Builds is not supported yet. We're working hard to deliver this soon!
+
+In the meantime, if you wish to build a managed project, you'll have to run `expo eject`. [Learn more here.](../../workflow/customizing/)
+
+## 5. If Building for Android...
+
+If you wish to build your application for Android but this is the first doing this, you'll have to generate a keystore. To do so, you'll need the `keytool` command installed and searchable in `PATH`. `keytool` is distributed with Java Development Kit. We recommend you [install JDK version 8](https://jdk.java.net/) as this specific version is needed for building Android apps.
```
Comment (cruzach) [inline docs/pages/build/setup.md:50]:
```suggestion
If you're building your Android application for the first time, you'll have to generate a keystore. To do so, you'll need the `keytool` command installed and searchable in `PATH`. `keytool` is distributed with Java Development Kit. We recommend you [install JDK version 8](https://jdk.java.net/) as this specific version is needed for building Android apps.
```

File: docs/pages/build/setup.md [hunk 9]
```diff
@@ -2,4 +2,72 @@
...
+
+See the [Android Credentials](../advanced-credentials-configuration/#android-credentials) section in the **Advanced Credentials Configuration** page to learn more about Android credentials.
+
+## 6. If Building for iOS...
+
+You’ll need a **paid** [Apple Developer Account](https://developer.apple.com/programs) to configure credentials required for the development and distribution process of an app.
+
+See the [iOS Credentials](../advanced-credentials-configuration/#ios-credentials) section in the **Advanced Credentials Configuration** page to learn more about iOS credentials.
+
+## 7. Run The Build
```
Comment (cruzach) [inline docs/pages/build/setup.md:60]:
```suggestion
## 7. Run the Build
```

File: docs/pages/build/setup.md [hunk 10]
```diff
@@ -2,4 +2,72 @@
...
+
+Run `expo eas:build --platform android` to build for Android, `expo eas:build --platform ios` to build for iOS, or `expo eas:build --platform all` to build for both platforms.
+
+## 8. Check the Status of Your Builds
+
+By default, the `expo eas:build` command is waiting for your build to complete. However, if you interrupt this command you can still monitor the progress of your builds by either visiting [the Expo website](https://expo.io/) or running the `expo eas:build:status` command.
```
Comment (cruzach) [inline docs/pages/build/setup.md:66]:
```suggestion
By default, the `expo eas:build` command will wait for your build to complete. However, if you interrupt this command you can still monitor the progress of your builds by either visiting [the Expo website](https://expo.io/) or running the `expo eas:build:status` command.
```

File: docs/pages/build/setup.md [hunk 11]
```diff
@@ -2,4 +2,72 @@
...
+
+## 9. See Other Docs and Stay Tuned
+
+- Read the [Configuring with eas.json](../eas-json/) guide to get familiar with EAS Builds configuration options.
+- If you want to learn more about internals of Android and iOS builds, check out [Android Builds](../android-builds/) and [iOS Builds](../ios-builds/) pages.
```
Comment (cruzach) [inline docs/pages/build/setup.md:71]:
```suggestion
- If you want to learn more about the internals of Android and iOS builds, check out our [Android Builds](../android-builds/) and [iOS Builds](../ios-builds/) pages.
```

File: docs/pages/build/setup.md [hunk 12]
```diff
@@ -2,4 +2,72 @@
...
+- Lastly, if you're feeling an expert on the credentials configuration, see the [Advanced Credentials Configuration](../advanced-credentials-configuration/) guide to customize the build process even further.
```
Comment (cruzach) [inline docs/pages/build/setup.md:72]:
```suggestion
- Lastly, if you feel like an expert on credentials configuration, see the [Advanced Credentials Configuration](../advanced-credentials-configuration/) guide to customize the build process even further.
```

File: docs/pages/build/setup.md [hunk 13]
```diff
@@ -2,4 +2,72 @@
...
+
+## 7. Run The Build
+
+Run `expo eas:build --platform android` to build for Android, `expo eas:build --platform ios` to build for iOS, or `expo eas:build --platform all` to build for both platforms.
+
+## 8. Check the Status of Your Builds
+
+By default, the `expo eas:build` command is waiting for your build to complete. However, if you interrupt this command you can still monitor the progress of your builds by either visiting [the Expo website](https://expo.io/) or running the `expo eas:build:status` command.
+
+## 9. See Other Docs and Stay Tuned
```
Comment (cruzach) [inline docs/pages/build/setup.md:68]:
```suggestion
## 9. Learn More
```

File: docs/pages/build/setup.md [hunk 14]
```diff
@@ -2,4 +2,72 @@
...
+        "workflow": "generic"
+      }
+    }
+  }
+}
+```
+
+If you want to learn more about the configuration options see the [Configuring with eas.json](../eas-json/) page.
+
+## 4. Eject Managed Project
```
Comment (cruzach) [inline docs/pages/build/setup.md:40]:
```suggestion
## 4. Eject Your Managed Project
```

File: docs/pages/build/android-builds.md [hunk 15]
```diff
@@ -75,26 +75,31 @@ android {
 }
 
 project.afterEvaluate {
-  android.signingConfigs.release { config ->
+  /* @info configure the release signing config */android.signingConfigs.release/* @end */ { config ->
```
Comment (cruzach) [inline docs/pages/build/android-builds.md:78]:
```suggestion
  /* @info This is where you configure the release signing config */android.signingConfigs.release/* @end */ { config ->
```

File: docs/pages/build/android-builds.md [hunk 16]
```diff
@@ -75,26 +75,31 @@ android {
...
     def debug = gradle.startParameter.taskNames.any { it.toLowerCase().contains('debug') }
 
+    /* @info don't read credentials.json if the task name contains debug */
```
Comment (cruzach) [inline docs/pages/build/android-builds.md:81]:
```suggestion
    /* @info Don't read credentials.json if the task name contains debug */
```

File: docs/pages/build/android-builds.md [hunk 17]
```diff
@@ -75,26 +75,31 @@ android {
...
     if (debug) {
       return
     }
+    /* @end */
 
     def credentialsJson = rootProject.file("../credentials.json");
 
     if (credentialsJson.exists()) {
+      /* @info don't do anything if the release signing config is already defined in build.gradle */
```
Comment (cruzach) [inline docs/pages/build/android-builds.md:90]:
```suggestion
      /* @info Don't do anything if the release signing config is already defined in build.gradle */
```

File: docs/pages/build/android-builds.md [hunk 18]
```diff
@@ -75,26 +75,31 @@ android {
...
       if (config.storeFile) {
         println("Path to release keystore file is already set, ignoring 'credentials.json'")
-      } else {
+      }/* @end */ else {
         try {
           def credentials = new groovy.json.JsonSlurper().parse(credentialsJson)
 
+          /* @info configure the signing config with the credentials.json data */
```
Comment (cruzach) [inline docs/pages/build/android-builds.md:97]:
```suggestion
          /* @info Use the data from credentials.json for the signing config */
```

File: docs/pages/build/android-builds.md [hunk 19]
```diff
@@ -106,9 +111,11 @@ project.afterEvaluate {
     }
   }
 
+  /* @info use the above signing config for the release build type */
```
Comment (cruzach) [inline docs/pages/build/android-builds.md:114]:
```suggestion
  /* @info Use the above signing config for the release build type */
```

**Review by cruzach — CHANGES_REQUESTED** [review]

File: docs/pages/build/android-builds.md [hunk 20]
```diff
@@ -75,26 +75,31 @@ android {
 }
 
 project.afterEvaluate {
-  android.signingConfigs.release { config ->
+  /* @info This is where you configure the release signing config */android.signingConfigs.release/* @end */ { config ->
```
Comment (dsokal) [inline docs/pages/build/android-builds.md:78]:
```suggestion
  /* @info This is where we configure the release signing config */android.signingConfigs.release/* @end */ { config ->
```

**Comment by dsokal** [comment]
I addressed all comments from Charlie and I'm merging this PR. If anyone else has more comments please let me know or open a PR :)

**Pull request merged**